cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/PublicationDaoImpl.java

```diff
@@ -67,11 +70,23 @@
         return results;
     }
 
+    @Override
+    public Page<PublicationImpl> getPublications(Pageable pageable) {
+
+        Query query = new Query().with(pageable);
+
+        List<PublicationImpl> list = mongoTemplate.find(query, PublicationImpl.class);
```

Comment:
I didn't catch that before, but this is probably not a good idea. You're loading *all* publications into memory and then take just one page. This will probably not work well for long (especially if more than one person is on the website). I would using a closableiterator here, which should be available on the mongoTemplate (https://docs.spring.io/spring-data/mongodb/docs/current/api/org/springframework/data/mongodb/core/MongoTemplate.html). You can use that directly or use `StreamUtils` (https://docs.spring.io/spring-data/data-commons/docs/current/api/org/springframework/data/util/StreamUtils.html) to turn it into a stream, which as a `skip` method. I believe (not 100% sure but this: https://stackoverflow.com/questions/11046105/iterate-over-large-collection-in-mongodb-via-spring-data/36967466, sounds like it), that by using streams a mongodb cursor will be used which will not have to iterate over all previous documents.